Resources/public/js/components/products/components/pricing/main.js

```diff
@@ -21,9 +21,17 @@ define(['config'], function (Config) {
 
             initForm.call(this, this.options.data);
 
+            var PriceListData = [];
```

Comment:
this can not work. As this is a list you can not add PriceListData.prices to it below. 
either you want a list and push a new item into it 

or you want to have an object ```{}``` and add a new property to it as you are currently doing it below